### PR TITLE
Use nireports for HTML reports

### LINKED ISSUE #805: Use nireports for report generation
## Summary

I'd like to do this right after we merge #802.

## Additional details

Synchrony with other preps.

### PR DESCRIPTION
**Pull request by tsalo**
Closes #805.

### CODE REVIEW

**Comment by mattcieslak** [comment]
I think we should move the interactive report workflows into a recon workflow instead of in reports

**Comment by tsalo** [comment]
Ah okay. I can drop it and just modify the expected outputs.

File: qsiprep/cli/run.py [hunk 1]
```diff
@@ -171,16 +171,16 @@ def main():
...
     finally:
 
-        from ..viz.reports import generate_reports
+        from ..reports.core import generate_reports
 
         # Generate reports phase
-        # session_list = (
-        #     config.execution.get().get('bids_filters', {}).get('dwi', {}).get('session')
-        # )
+        session_list = config.execution.get().get("bids_filters", {}).get("dwi", {}).get("session")
```
Comment (mattcieslak) [inline qsiprep/cli/run.py:177]: this is some interesting syntax

File: qsiprep/config.py [hunk 2]
```diff
@@ -433,6 +433,7 @@ class execution(_Config):
     """Write out the computational graph corresponding to the planned preprocessing."""
     dataset_links = {}
     """A dictionary of dataset links to be used to track Sources in sidecars."""
+    aggr_ses_reports = 4
```
Comment (mattcieslak) [inline qsiprep/config.py:436]: does this work now?

**Review by mattcieslak — APPROVED** [review]
Does this nireports release support gifs?

**Comment by tsalo** [comment]
Unfortunately not yet. We need to follow up with nireports to support gifs.

**Comment by tsalo** [comment]
I've opened a PR to support GIFs in https://github.com/nipreps/nireports/pull/128.

**Review by mattcieslak — APPROVED** [review]

**Pull request merged**